### ISSUE TITLE
bug no-browser-hacks with scss variables

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#871]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Rule `no-browser-hacks` bugs with scss variables.

> Which rule, if any, is this issue related to?

`no-browser-hacks`

> What CSS is needed to reproduce this issue?

```scss
$xs-and-up: "(min-width:" $screen-xs ")";
$sm-and-up: "(min-width:" $screen-sm ")";
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "no-browser-hacks": true
  }
}
```

> Which version of stylelint are you using?

e.g. `4.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

SCSS

> What did you expect to happen?

No errors.

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
_variables.scss
2:1     Unexpected property hack "$sm-and-up" (no-browser-hacks)
```

**Comment by jeddy3** [comment]
@davidtheclark & @ben-eb: how shall we handle these issues with rules that wrap 3rd party modules?

We can't force the same approach we take here of ignoring non-standard CSS syntax on these modules. Especially as something like this will require a bit of thinking i.e. an old IE 5 hack has the same signature as SCSS variables.

I guess the simplest change (and one we can do right now) is a documentation one... something along the lines of "As this rule uses `stylehacks` under the hood is is *only* compatible with standard CSS syntax."? Seem fair to me.

**Comment by davidtheclark** [comment]
Yes, I think that a statement like that would help in the docs. Also we could link to the Issues tab of the given repo so that users can file bugs and feature requests there.

**Comment by ben-eb** [comment]
Not sure to be honest; like you said this only applies to Sass variables and, if it were to be disabled, would result in this hack not being caught. Are there any other instance that you can think of that the default behaviour is not appropriate for Sass linting?

**Comment by jeddy3** [comment]
> Are there any other instances that you can think of where the default behaviour is not appropriate for Sass linting?

I'm not sure I'm afraid as I not too much familiar with all the the browser hacks that stylehacks checks.

It's documented as being compatible only with standard CSS syntax now. Perhaps we can leave it at that for now?

**Comment by ArmorDarks** [comment]
What about ability to ignore some hacks? For example something like `ignore: [^\$.*$]` should allow to use it with Sass variables. Yes, some hacks might not be covered with such ignore, but it's better than nothing. Especially considering that most IE hacks using `*` or `_`, but not `$`

**Comment by ben-eb** [comment]
I've got an open issue for adding optional Sass support to stylehacks. I'm more than happy if someone wants to maintain this support, but I don't use Sass these days so it's not something that I'm going to add myself.

https://github.com/ben-eb/stylehacks/issues/21
